types/leaflet/index.d.ts

```diff
@@ -668,6 +668,12 @@ export function canvas(options?: RendererOptions): Canvas;
  */
 export class LayerGroup<P = any> extends Layer {
     constructor(layers?: Layer[]);
+    // tslint:disable-next-line
+    constructor(layers: Layer[], options?: LayerOptions);
+    initialize(layers?: Layer[]): this;
```

Comment:
`initialize` is an internal method to provide a constructor within Leaflet's class `Class`. You could add this to Leaflet's `Class` class, but I would prefer to put this not into the type-definition. What is your opinion on that @alejo90?

Reply:
I agree with you on that one. It would be better to add it to `Class`, but better still to not have it at all, since everything can now be done via `extends`.

Reply:
In my opinion I would prefer not to add `initialize` to the definition. At first you can work with the `class` and `extends` keywords and just call `super`. Then you use the "normal way" of inheritance in TypeScript.

Example:

```typescript
class MyExtendedLayerGroup extends L.LayerGroup {
  constructor(options) {
    super(options);
    L.setOptions(this, options);
  }
}
```

In addition `initialize` is [not a documented method of Leaflet classes](http://leafletjs.com/reference-1.2.0.html). In my opinion it is something like a protected method, that has no effort in TypeScript.

Reply:
I completely agree. `initialize` is probably stil in there as a legacy way of doing things and because of the effor it would take to remove it altogether from the leaflet codebase. That being said, it doesn't make sense in typescript anymore. So, the deeper question is: should we disencourage it by removing `Class` from the definitions or leave it as an `use-at-your-own-risk` option and add the `initialize` metod to it?